server/app/graph.go

```diff
@@ -26,3 +29,42 @@ func (a *App) GetFrontEndGraph() *model.FrontendGraph {
 
 	return &gr
 }
+
+func (a *App) GetNextNodes(userID string) ([]model.Node, []model.Node, error) {
+	statuses, err := a.GetStatusesForUser(userID)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "can't get statuses for user %s", userID)
+	}
+	statusMap := map[string]*model.NodeStatusForUser{}
+	for _, status := range statuses {
+		statusMap[status.NodeID] = status
+	}
+
+	inProgressNodes := []model.Node{}
+	nextNodes := []model.Node{}
+
+	for _, node := range a.Graph.Nodes {
+		status, ok := statusMap[node.ID]
+		if ok && (status.Status == model.NodeStatusStarted || status.Status == model.NodeStatusWatched) {
+			inProgressNodes = append(inProgressNodes, node)
+		} else if !ok || status.Status != model.NodeStatusFinished {
+			prereqs, ok2 := a.Graph.Prerequisites[node.ID]
+			if !ok2 {
+				return nil, nil, errors.Wrapf(err, "no prerequisites for node %s", node.ID)
+			}
+			isNext := true
+			for _, prereq := range prereqs {
+				prereqStatus, ok3 := statusMap[prereq]
+				if !ok3 || prereqStatus.Status != model.NodeStatusFinished {
+					isNext = false
+					break
+				}
+			}
+
+			if isNext || !ok {
```

Comment:
the `|| !ok` makes this code a little difficult to understand. Would it be better to move it after line 47:
```
status, ok := statusMap[node.ID]
if !ok {
...
continue
if (status.Status == model.NodeStatusStarted || status.Status == model.NodeStatusWatched) {
...
} else if status.Status != model.NodeStatusFinished {
...
}
```